client/src/ui/ToasterWebview.ts

```diff
@@ -0,0 +1,202 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) 2023 Savoir-faire Linux. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode'
+import type { IPty } from 'node-pty'
+import { BitbakeTerminal, bitbakeTerminals } from './BitbakeTerminal'
+import { BitbakeDriver } from '../driver/BitbakeDriver'
+import { pty } from '../utils/ProcessUtils'
+import { logger } from '../lib/src/utils/OutputLogger'
+import { loadBitbakeSettings } from '../lib/src/BitbakeSettings'
+
+const DEFAULT_TOASTER_PORT = 8000
+
+export class ToasterPanel {
+  /**
+  * Track the currently panel. Only allow a single panel to exist at a time.
+  */
+  public static currentPanel: ToasterPanel | undefined
+  public static toasterProcess: IPty | undefined
+  public static readonly viewType = 'bitbake.toasterView'
+
+  private readonly _panel: vscode.WebviewPanel
+  private readonly _disposables: vscode.Disposable[] = []
+
+  public static createOrShow (): void {
+    const column = vscode.window.activeTextEditor !== undefined
+      ? vscode.window.activeTextEditor.viewColumn
+      : undefined
+
+    // If we already have a panel, show it.
+    if (ToasterPanel.currentPanel !== undefined) {
+      ToasterPanel.currentPanel._panel.reveal(column)
+      return
+    }
+
+    // Otherwise, create a new panel.
+    const panel = vscode.window.createWebviewPanel(
+      ToasterPanel.viewType,
+      'Yocto Toaster',
+      column ?? vscode.ViewColumn.One,
+      getWebviewOptions()
+    )
+
+    ToasterPanel.currentPanel = new ToasterPanel(panel)
+  }
+
+  public static async startToasterInBrowser (): Promise<void> {
+    await ToasterPanel.runToasterCommand('source toaster start', true)
+    const url = `http://localhost:${DEFAULT_TOASTER_PORT}`
+    await vscode.env.openExternal(vscode.Uri.parse(url)).then(success => {
+      if (!success) {
+        void vscode.window.showErrorMessage(`Failed to open URL ${url}`)
+      }
+    })
+  }
+
+  // Close the webview and stop the toaster
+  public static stop (): void {
+    ToasterPanel.currentPanel?.dispose()
+  }
+
+  // A custom replicate of runBitbakeCommand() from BitbakeTerminal.ts
+  private static async runToasterCommand (command: string, isBackground: boolean = false, terminalName: string = 'Toaster'): Promise<IPty> {
```

Comment:
Avoid code duplication, what was missing on the original runBitbakeCommand?

Reply:
Not missing, but it has more than what is needed here.